Import Essential Libraries

In [45]:
import pandas as pd
import sklearn as sk
import matplotlib.pyplot as plt
import numpy as np

Load the data and Check the info of data

In [2]:
data=pd.read_csv('uci-secom.csv')
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1567 entries, 0 to 1566
Columns: 592 entries, Time to Pass/Fail
dtypes: float64(590), int64(1), object(1)
memory usage: 7.1+ MB


In [22]:
data.shape

(592,)

In [5]:
data.head()

Time        0        1          2          3       4      5  \
0  2008-07-19 11:55:00  3030.93  2564.00  2187.7333  1411.1265  1.3602  100.0   
1  2008-07-19 12:32:00  3095.78  2465.14  2230.4222  1463.6606  0.8294  100.0   
2  2008-07-19 13:17:00  2932.61  2559.94  2186.4111  1698.0172  1.5102  100.0   
3  2008-07-19 14:43:00  2988.72  2479.90  2199.0333   909.7926  1.3204  100.0   
4  2008-07-19 15:22:00  3032.24  2502.87  2233.3667  1326.5200  1.5334  100.0   

          6       7       8  ...       581     582     583     584      585  \
0   97.6133  0.1242  1.5005  ...       NaN  0.5005  0.0118  0.0035   2.3630   
1  102.3433  0.1247  1.4966  ...  208.2045  0.5019  0.0223  0.0055   4.4447   
2   95.4878  0.1241  1.4436  ...   82.8602  0.4958  0.0157  0.0039   3.1745   
3  104.2367  0.1217  1.4882  ...   73.8432  0.4990  0.0103  0.0025   2.0544   
4  100.3967  0.1235  1.5031  ...       NaN  0.4800  0.4766  0.1045  99.3032   

      586     587     588       589  Pass/Fail  
0     NaN     NaN     NaN       NaN         -1  
1  0.0096  0.0201  0.0060  208.2045         -1  
2  0.0584  0.0484  0.0148   82.8602          1  
3  0.0202  0.0149  0.0044   73.8432         -1  
4  0.0202  0.0149  0.0044   73.8432         -1  

[5 rows x 592 columns]

The data consists of  1567 entries with 592 features, which it would mean

In [7]:
data.isnull().sum()

Time          0
0             6
1             7
2            14
3            14
             ..
586           1
587           1
588           1
589           1
Pass/Fail     0
Length: 592, dtype: int64

In [14]:
pd.set_option('display.max_rows',None)

In [23]:
no_na=data.isnull().sum()
pd.DataFrame(data=no_na,index=data.columns)

0
Time          0
0             6
1             7
2            14
3            14
4            14
5            14
6            14
7             9
8             2
9             2
10            2
11            2
12            2
13            3
14            3
15            3
16            3
17            3
18            3
19           10
20            0
21            2
22            2
23            2
24            2
25            2
26            2
27            2
28            2
29            2
30            2
31            2
32            1
33            1
34            1
35            1
36            1
37            1
38            1
39            1
40           24
41           24
42            1
43            1
44            1
45            1
46            1
47            1
48            1
49            1
50            1
51            1
52            1
53            4
54            4
55            4
56            4
57            4
58            4
59            7
60            6
61            6
62            6
63            7
64            7
65            7
66            6
67            6
68            6
69            6
70            6
71            6
72          794
73          794
74            6
75           24
76           24
77           24
78           24
79           24
80           24
81           24
82           24
83            1
84           12
85         1341
86            0
87            0
88            0
89           51
90           51
91            6
92            2
93            2
94            6
95            6
96            6
97            6
98            6
99            6
100           6
101           6
102           6
103           2
104           2
105           6
106           6
107           6
108           6
109        1018
110        1018
111        1018
112         715
113           0
114           0
115           0
116           0
117           0
118          24
119           0
120           0
121           9
122           9
123           9
124           9
125           9
126           9
127           9
128           9
129           9
130           9
131           9
132           8
133           8
134           8
135           5
136           6
137           7
138          14
139          14
140          14
141          14
142          14
143           9
144           2
145           2
146           2
147           2
148           2
149           3
150           3
151           3
152           3
153           3
154           3
155          10
156           0
157        1429
158        1429
159           2
160           2
161           2
162           2
163           2
164           2
165           2
166           2
167           2
168           2
169           2
170           1
171           1
172           1
173           1
174           1
175           1
176           1
177           1
178          24
179           1
180           1
181           1
182           1
183           1
184           1
185           1
186           1
187           1
188           1
189           1
190           4
191           4
192           4
193           4
194           4
195           4
196           7
197           6
198           6
199           6
200           7
201           7
202           7
203           6
204           6
205           6
206           6
207           6
208           6
209           6
210          24
211          24
212          24
213          24
214          24
215          24
216          24
217          24
218           1
219          12
220        1341
221           0
222           0
223           0
224          51
225          51
226           6
227           2
228           2
229           6
230           6
231           6
232           6
233           6
234           6
235           6
236           6
237           6
238           2
239           2
240           6
241           6
242           6
243           6
244        1018
245        1018
246        1018
247         715
248           

In [25]:
no_na.sort_values(ascending=False)

158          1429
292          1429
293          1429
157          1429
85           1341
492          1341
220          1341
358          1341
517          1018
245          1018
246          1018
518          1018
382          1018
516          1018
109          1018
111          1018
244          1018
110          1018
384          1018
383          1018
579           949
581           949
580           949
578           949
346           794
345           794
73            794
72            794
519           715
385           715
112           715
247           715
568           273
566           273
565           273
569           273
563           273
562           273
567           273
564           273
555           260
553           260
554           260
557           260
556           260
549           260
546           260
548           260
547           260
550           260
551           260
552           260
90             51
89             51
363            51
362       

In [37]:
typedata={str(k): len(list(v)) for k, v in data.groupby(data.dtypes, axis=1)}
typedata

{'int64': 1, 'float64': 590, 'object': 1}

In [44]:
data.groupby(data.dtypes,axis=1).groups

{int64: ['Pass/Fail'], float64: ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10', '11', '12', '13', '14', '15', '16', '17', '18', '19', '20', '21', '22', '23', '24', '25', '26', '27', '28', '29', '30', '31', '32', '33', '34', '35', '36', '37', '38', '39', '40', '41', '42', '43', '44', '45', '46', '47', '48', '49', '50', '51', '52', '53', '54', '55', '56', '57', '58', '59', '60', '61', '62', '63', '64', '65', '66', '67', '68', '69', '70', '71', '72', '73', '74', '75', '76', '77', '78', '79', '80', '81', '82', '83', '84', '85', '86', '87', '88', '89', '90', '91', '92', '93', '94', '95', '96', '97', '98', '99', ...], object: ['Time']}